okhttp/src/test/java/io/grpc/okhttp/OkHttpServerTransportTest.java

```diff
@@ -993,6 +1000,59 @@ public void httpErrorsAdhereToFlowControl() throws Exception {
     shutdownAndTerminate(/*lastStreamId=*/ 1);
   }
 
+  @Test
+  public void windowUpdate() throws Exception {
+    initTransport();
+    handshake();
+    OkHttpServerTransport.FrameHandler handler = serverTransport.getHandler();
```

Comment:
I really don't want to access this handler directly, as it changes the threading model. Did you see how difficult it'd be to manually encode a data frame with padding?

Reply:
I don't think we need to change `new Http2().newWriter`. We just need to write directly to the `socketPair.getClientOutputStream()` for the data frame with padding.